front/src/applications/stdcm/utils/formatStdcmConfV2.ts

```diff
@@ -137,9 +138,36 @@ export const checkStdcmConf = (
         metadata,
         ...path
       } = step;
+
+      const isTrackOffset = (steps: PathStep): steps is TrackOffset & PathStep =>
+        'trackId' in steps && 'offset' in steps;
+      const isOperationalPoint = (
+        steps: PathStep
+      ): steps is { operational_point: string } & PathStep => 'operational_point' in steps;
+      const isTrigramType = (
+        steps: PathStep
+      ): steps is { secondary_code: string | null; trigram: string } & PathStep =>
+        'trigram' in steps;
+      const isUicType = (
+        steps: PathStep
+      ): steps is { secondary_code: string | null; uic: number } & PathStep => 'uic' in steps;
+
+      let location;
+      if (isTrackOffset(step)) {
+        location = step;
+      } else if (isOperationalPoint(step)) {
+        location = { operational_point: step.operational_point };
+      } else if (isTrigramType(step)) {
+        location = { secondary_code: ch ?? null, trigram: step.trigram };
+      } else if (isUicType(step)) {
+        location = { secondary_code: ch ?? null, uic: step.uic };
+      } else {
+        location = { ...path, secondary_code: ch };
+      }
```

Comment:
Actually, you can do it more simply :) You can remove all the unwanted props from step l140 (you need to add theoreticalMargin, kp and onStopSignal) and then you can just define secondary code if the path step location has trigram or uic.
```suggestion
const secondary_code = 'trigram' in path || 'uic' in path ? ch : undefined;
```
You can also rename path by stepLocation or location

Reply:
Nice catch ! done